openpype/hosts/maya/plugins/publish/collect_animation.py

```diff
@@ -58,17 +58,30 @@ def process(self, instance):
         if instance.data.get("farm"):
             instance.data["families"].append("publish.farm")
 
-        # Collect user defined attributes.
-        if not instance.data.get("includeUserDefinedAttributes", False):
-            return
-
-        user_defined_attributes = set()
-        for node in hierarchy:
-            attrs = cmds.listAttr(node, userDefined=True) or list()
-            shapes = cmds.listRelatives(node, shapes=True) or list()
-            for shape in shapes:
-                attrs.extend(cmds.listAttr(shape, userDefined=True) or list())
+        # Backwards compatibility for attributes.
+        backwards_mapping = {
+            "write_color_sets": "writeColorSets",
+            "write_face_sets": "writeFaceSets",
+            "include_parent_hierarchy": "includeParentHierarchy",
+            "include_user_defined_attributes": "includeUserDefinedAttributes"
+        }
+        for key, value in backwards_mapping.items():
+            if key in instance.data:
+                instance.data[value] = instance.data[key]
 
-            user_defined_attributes.update(attrs)
-
-        instance.data["userDefinedAttributes"] = list(user_defined_attributes)
+        # Collect user defined attributes.
+        if instance.data.get("includeUserDefinedAttributes", False):
+            user_defined_attributes = set()
+            for node in hierarchy:
+                attrs = cmds.listAttr(node, userDefined=True) or list()
+                shapes = cmds.listRelatives(node, shapes=True) or list()
+                for shape in shapes:
+                    attrs.extend(
+                        cmds.listAttr(shape, userDefined=True) or list()
+                    )
```

Comment:
This seems to already be in the original code. But isn't this wrong (or doubling up the data) since basically `hierarchy` that we're iterating already includes the descendents including the shapes. 

Also, side note if we do need to keep it for some reason then `listRelatives` MUST use the `fullPath=True` argument to avoid name clashes for shapes that do not have a unique name. I have a feeling this will fail already on instances AND on `["cube1|shape", "cube2|shape"]`.